design-docs/001-virtual-chunk-improvements.md

```diff
@@ -138,9 +138,9 @@ When a virtual chunk is requested, Icechunk will find in the `location` field of
 
 The mapping is done by comparing the protocol and path to the set of configured virtual chunk containers, plus a set of default virtual chunk containers.
 
-This is how the mapping from location to cloud+bucket+prefix+credentials works:
+This is how the mapping from a chunk location (url) to cloud+bucket+prefix+credentials works:
 
-* Using the `location` protocol we find all containers that match the protocol
+* Using the chunk location's protocol, we find all containers that have a matching `protocol` field
```

Comment:
It's just the first few characters of the prefix - why separate it out from the prefix in general? Sometimes the prefix might be "s3", sometimes "s3://bucket", sometimes "s3://bucket/path/to". Aside from knowing that these particular examples are all s3 protocol, I don't think that there's any difference in how you should treat them.

Reply:
I like this. My remaining concern is, how to make it simple for the simple use case. Say, virtual chunks in google storage, repository in AWS. I don't want users to _have to_ create a container for this simple case. That's why I'm proposing to distinguish the platform by the url protocol (by default).

Reply:
Then you have two:
- "s3"
- "gs"
(and they happen to exactly look like what you would expect as protocols).

You could even have the default default be "", i.e., matches any prefix that's not matched by something more specific.